src/foam/java/refinements.js

```diff
@@ -321,6 +321,34 @@ foam.CLASS({
   ]
 });
 
+foam.CLASS({
+  refines: 'foam.core.Action',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'javaCode'
+    },
+    {
+      class: 'String',
+      name: 'javaReturns'
+    }
+  ],
+
+  methods: [
+    function buildJavaClass(cls) {
+      if ( ! this.javaCode ) return;
+
+      cls.method({
+        visibility: 'public',
+        name: this.name,
+        type: 'void',
+        static: true,
```

Comment:
I don't think the method should be static, we want to apply it to instances of the class we are generating.